diff --git a/flag.tokens.inc b/flag.tokens.inc
index cae4d88..ed72ccf 100644
--- a/flag.tokens.inc
+++ b/flag.tokens.inc
@@ -83,21 +83,24 @@ function flag_token_info() {
   if (module_exists('token')) {
     $entity_info = entity_get_info();
     foreach (flag_get_types() as $flag_type) {
-      // The flag type is the entity type, but this is not necessarily the same
-      // as the entity's token type.
-      $token_type = $entity_info[$flag_type]['token type'];
-      $flags = flag_get_flags($flag_type);
-      foreach ($flags as $flag) {
-        $tokens[$token_type]['flag-' . str_replace('_', '-', $flag->name) . '-count'] = array(
-          'name' => t('@flag flag count', array('@flag' => $flag->get_title())),
-          'description' => t('Total flag count for flag @flag', array('@flag' => $flag->get_title())),
-          'flag-type' => $flag_type,
-        );
-        $tokens[$token_type]['flag-' . str_replace('_', '-', $flag->name) . '-link'] = array(
-          'name' => t('@flag flag link', array('@flag' => $flag->get_title())),
-          'description' => t('Flag/unflag link for @flag', array('@flag' => $flag->get_title())),
-          'flag-type' => $flag_type,
-        );
+      if (!empty($entity_info[$flag_type]['token type'])) {
+
+        // The flag type is the entity type, but this is not necessarily the same
+        // as the entity's token type.
+        $token_type = $entity_info[$flag_type]['token type'];
+        $flags = flag_get_flags($flag_type);
+        foreach ($flags as $flag) {
+          $tokens[$token_type]['flag-' . str_replace('_', '-', $flag->name) . '-count'] = array(
+            'name' => t('@flag flag count', array('@flag' => $flag->get_title())),
+            'description' => t('Total flag count for flag @flag', array('@flag' => $flag->get_title())),
+            'flag-type' => $flag_type,
+          );
+          $tokens[$token_type]['flag-' . str_replace('_', '-', $flag->name) . '-link'] = array(
+            'name' => t('@flag flag link', array('@flag' => $flag->get_title())),
+            'description' => t('Flag/unflag link for @flag', array('@flag' => $flag->get_title())),
+            'flag-type' => $flag_type,
+          );
+        }
       }
     }
   }
